config/mbed/CMakeLists.txt

```diff
@@ -0,0 +1,279 @@
+list(APPEND CHIP_CFLAGS ${CHIP_MBEDSTORAGE_CFLAGS})
+
+# Add support for Mbed Socket 
+mbed_get_target_common_compile_flags(CHIP_MBEDNETSOCKET_CFLAGS mbed-netsocket)
+list(APPEND CHIP_CFLAGS ${CHIP_MBEDNETSOCKET_CFLAGS})
+
+if (CONFIG_CHIP_WITH_EXTERNAL_MBEDTLS)
+    mbed_get_target_common_compile_flags(CHIP_MBEDTLS_CFLAGS mbed-mbedtls)
+    list(APPEND CHIP_CFLAGS ${CHIP_MBEDTLS_CFLAGS})
+    if (${MBED_TARGET} MATCHES "NRF52840_DK")
+        mbed_get_target_common_compile_flags(CHIP_CRYPTOCELL310_CFLAGS mbed-mbedtls-cryptocell310)
+        list(APPEND CHIP_CFLAGS ${CHIP_CRYPTOCELL310_CFLAGS})
+    endif()
+endif()
+
+list(APPEND CHIP_CFLAGS 
+    \"-D__LINUX_ERRNO_EXTENSIONS__=1\"
+)
+
+set(SEPARATOR ",")
+convert_list_of_flags_to_string_of_flags(CHIP_CFLAGS CHIP_CFLAGS ${SEPARATOR})
+set(SEPARATOR " ")
+convert_list_of_flags_to_string_of_flags(CHIP_CFLAGS_C CHIP_CFLAGS_C ${SEPARATOR})
+convert_list_of_flags_to_string_of_flags(CHIP_CFLAGS_CC CHIP_CFLAGS_CC ${SEPARATOR})
+
+# Prepare CHIP libraries that the application should be linked with
+
+if (NOT CHIP_LIBRARIES)
+    set(CHIP_LIBRARIES -lCHIP)
+endif()
+
+if (CONFIG_CHIP_LIB_SHELL)
+    list(APPEND CHIP_LIBRARIES -lCHIPShell)
+endif()
+
+# Set up CHIP project configuration file
+
+set(CHIP_DEFAULT_CONFIG_FILE "<${CHIP_ROOT}/config/mbed/CHIPProjectConfig.h>")
+set(CHIP_PROJECT_CONFIG         ${CHIP_DEFAULT_CONFIG_FILE})
+
+if (CONFIG_CHIP_PROJECT_CONFIG)
+    get_filename_component(CHIP_PROJECT_CONFIG 
+        ${CONFIG_CHIP_PROJECT_CONFIG}
+        REALPATH
+        BASE_DIR ${CMAKE_SOURCE_DIR}
+    )
+    set(CHIP_PROJECT_CONFIG "<${CHIP_PROJECT_CONFIG}>")
+endif()
+
+if (${CMAKE_BUILD_TYPE} STREQUAL "debug")
+    set(CONFIG_DEBUG "y")
+endif()
+
+# ==============================================================================
+# Generate configuration for CHIP GN build system
+# ==============================================================================
+
+chip_gn_arg_flags("target_cflags"                         ${CHIP_CFLAGS})
+chip_gn_arg_lang_flags("target_cflags_c"                  ${CHIP_CFLAGS_C})
+chip_gn_arg_lang_flags("target_cflags_cc"                 ${CHIP_CFLAGS_CC})
+chip_gn_arg_string("mbed_ar"                              ${CMAKE_AR})
+chip_gn_arg_string("mbed_cc"                              ${CMAKE_C_COMPILER})
+chip_gn_arg_string("mbed_cxx"                             ${CMAKE_CXX_COMPILER})
+chip_gn_arg_string("chip_project_config_include"          "${CHIP_PROJECT_CONFIG}")
+chip_gn_arg_bool  ("is_debug"                             CONFIG_DEBUG)
+chip_gn_arg_bool  ("chip_build_tests"                     CONFIG_CHIP_BUILD_TESTS)
+chip_gn_arg_bool  ("chip_build_libshell"                  CONFIG_CHIP_LIB_SHELL)
+chip_gn_arg_bool  ("chip_with_platform_mbedtls"           CONFIG_CHIP_WITH_EXTERNAL_MBEDTLS)
+chip_gn_arg_bool  ("chip_bypass_rendezvous"               CONFIG_CHIP_BYPASS_RENDEZVOUS)
+
+if (BOARD STREQUAL "native_posix")
```

Comment:
Are those ifs valid for mbed-os as well or can be removed?

Reply:
Not applicable at the moment, I will remove and reintroduce at the right time.

Reply:
Removed